modules/standard/Regexp.chpl

```diff
@@ -452,7 +452,7 @@ class BadRegexpError : Error {
 
  */
 proc compile(pattern: ?t, posix=false, literal=false, noCapture=false,
-             /*i*/ ignoreCase=false, /*m*/ multiLine=false, /*s*/ dotnl=false,
+             /*i*/ ignoreCase=false, /*m*/ multiLine=false, /*s*/ dotAll=false,
```

Comment:
we can deprecate the `dotnl` argument by adding a `proc compile` with the old signature, adding `compilerWarning` call in it, and mark it with `pragma "last resort"`. The argument names are part of the public API since you can do things like `compile("s", dotnl=true)`.

When deprecating, we also add a test to test/deprecated.  See also https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst .